src/main/java/updater/JarReplacerAndExecutor.java

```diff
@@ -0,0 +1,231 @@
+package updater;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.control.Label;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+
+/**
+ * Note: This class will be compiled into a JAR on its own
+ *
+ * Replaces a JAR file and start that JAR
+ */
+public class JarReplacerAndExecutor extends Application{
```

Comment:
`And` in the class name is a sign of violating Single Responsibility Principle. :-)